discovery/azure/azure.go

```diff
@@ -305,6 +305,7 @@ type virtualMachine struct {
 	Location          string
 	OsType            string
 	ScaleSet          string
+	InstanceID        string
```

Comment:
This is great, but I think we need to update https://github.com/prometheus/prometheus/blob/818524882328307a761a5bd8091f829f572bada9/discovery/azure/azure.go#L569 as well to make this work?

Reply:
I don't think so - it should just return the zero-value (an empty string) in that field.  But if it makes it more readable to set it explicitly to an empty string, I can do that.

Reply:
no, its ok, I missed your change in  `mapFromVMScaleSetVM`-- all good!